### PR TITLE
Merge `Qual` and `Callers` cell on cancer SNVs page

### LINKED ISSUE #5043: Remove QUAL column on cancer SNVs now that callers are shown on Callers column
It's redundant info

### LINKED ISSUE #5044: Some cancer SNVs have filter but not callers
Like in [this situation](https://scout-stage.scilifelab.se/cust000/G1A1471p10_Balsamic/cancer/variants?variant_type=clinical). Here the callers cell is empty.

Perhaps refactor to merge call and filters, to solve both #5043 and this issue

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #5044 and closes #5043 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Check SNV variantS from [this case](https://scout-stage.scilifelab.se/cust000/G1A1471p10_Balsamic) but also other cases, cancer and non-cancer (only SVs in the latter case)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/variants/templates/variants/utils.html [hunk 1]
```diff
@@ -49,11 +49,17 @@
...
-  {% for name, caller in callers %}
+{% macro callers_cell(variant) %}
+  {% for name, caller in variant.callers %}
       <span class="badge {% if caller == 'Pass' %}bg-success{% elif caller == 'Filtered' %}bg-secondary{% else %}bg-black{% endif %}" data-bs-toggle="tooltip" data-bs-html="true" title="{{caller}}">
         {{ name }}
       </span>
+  {% else %} <!-- collect info from variant's filters -->
+    {% for filter in variant.filters %}
+      <span class="badge bg-{{ filter.label_class }}" data-bs-toggle="tooltip" data-bs-placement="top" title="{{ filter.description }}">
+        {{ filter.label }}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/utils.html:60]: On a cursory look, the filters are quite a bit richer, and has nicely templated categories. I feel when we have them we should show them? But, let's check a few files or discuss with the Balsamic team I guess!

**Comment by northwestwitch** [comment]
### Demo case:

image

### Test case on scout-stage:

image

**Review by dnil — APPROVED** [review]
This is better than current prod, and than current main, so clearly approved! It gets a little tedious with the redundant PASS and FAIL ones, so checking against that would be an improvement.

**Comment by northwestwitch** [comment]
> It gets a little tedious with the redundant PASS and FAIL ones, so checking against that would be an improvement.

You want me to add this check in the PR or it is fine as it is for now? Pehaps we could get some users' feedback afterwards

**Comment by dnil** [comment]
> > It gets a little tedious with the redundant PASS and FAIL ones, so checking against that would be an improvement.
> 
> You want me to add this check in the PR or it is fine as it is for now? Pehaps we could get some users' feedback afterwards

I think it is tentatively better than main already, so good to merge if you like. If it looks easy and safe to do the check, I'd say do it now while we are thinking about it. But if it gets complicated its fine to leave the extra badges there for a bit.

**Comment by northwestwitch** [comment]
> I think it is tentatively better than main already, so good to merge if you like. If it looks easy and safe to do the check, I'd say do it now while we are thinking about it. But if it gets complicated its fine to leave the extra badges there for a bit.

I'll just merge then. If sombody complains we can fix it

**Pull request merged**